### ISSUE TITLE
Import meta graph followed by save overwrites the previous checkpoints

### ISSUE THREAD

**Issue by ramnath-k** [tensorflow/tensorflow#7876]
NOTE: Only file GitHub issues for bugs and feature requests.  All other topics will be closed.

For general support from the community, see [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).
To make bugs and feature requests more easy to find and organize, we close issues that are deemed
out of scope for GitHub Issues and point people to StackOverflow.

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

### Environment info
Operating System: Ubuntu 14.04

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`): using CPU version of tensorflow 1.0.0

If installed from binary pip package, provide:

1. A link to the pip package you installed: pip install tensorflow
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`. 1.0.0

If installed from source, provide 

1. The commit hash (`git rev-parse HEAD`)
2. The output of `bazel version`

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
```python
import tensorflow as tf

graph = tf.Graph()
with graph.as_default():
    initializer = tf.random_uniform_initializer(minval=-0.5, maxval=0.5, seed=42, dtype=tf.float32)
    var1 = tf.get_variable('var1', shape=(1,), dtype=tf.float32, initializer=initializer)
    saver = tf.train.Saver(tf.trainable_variables(), max_to_keep=20)
    init_op = tf.global_variables_initializer()
    graph.finalize()

with tf.Session(graph=graph) as sess:
    sess.run(init_op)
    saver.save(sess, 'sample_graph', global_step=0)

graph = tf.Graph()
with tf.Session(graph=graph) as sess:
    saver = tf.train.import_meta_graph('sample_graph-0.meta')
    saver.restore(sess, './sample_graph-0')
    saver.save(sess, 'sample_graph', global_step=1)
    print(saver.last_checkpoints) # lists only ['sample_graph-1'] does not preserve the previous checkpoint sample_graph-0
```
Essentially I am checkpointing a graph and then importing it. On trying to save the next checkpoint, the saver overwrites the previous checkpoint in the checkpoint file (the actual meta, index and data files are not overwritten) and only the last saved checkpoint is present in the `checkpoint` file. Is this the intended behavior? Is there any way to preserve the checkpoints across multiple saves of the graph.

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

**Comment by poxvoculi** [comment]
Thanks for a clear issue report.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activityand the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by martinwicke** [comment]
@sherrym is this intended? Is the checkpoint still there but the checkpoint list is cleared since it's a new saver/graph?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 107 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 139 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 154 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 169 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 184 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 199 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 214 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 229 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 243 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 258 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sherrym: It has been 273 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by harshini-gadige** [comment]
Is this still an issue ?
@sherrym - Could you please look into this ?

**Comment by martinwicke** [comment]
Reassigning to Allen, Sherry has left the team. 

I think what is happening is that the list of checkpoints is reset by import_meta_graph. The old checkpoints still exist on disk, but the index structure is overwritten by import_meta_graph.

We won't do any work on this before 2.0, and I'm concerned that we won't be able to really fix this since it would be a behavior change (and it's not clear that this is a bug -- import_meta_graph often does invalidate old checkpoints since you may add new variables). 

I will close this issue. This should be much less of a problem in 2.0.

**Comment by allenlavoie** [comment]
There is a `write_state` Boolean argument to `Saver.save`, so you can prevent it from updating the Checkpoint file. And there is also `Saver.set_last_checkpoints_with_time`, which you can use along with `tf.train.get_checkpoint_mtimes` so that the `Saver` writes a correct-ish Checkpoint file instead of just not overwriting.

I do think the 2.x story will be better here. Still working out the details, but there should be a design review soon for importing SavedModels. I'll include an import-and-train workflow there.
